File: app/src/main/java/dev/ragnarok/fenrir/db/model/entity/feedback/CopyEntity.java
package dev.ragnarok.fenrir.db.model.entity.feedback;

import com.google.gson.annotations.SerializedName;

import dev.ragnarok.fenrir.db.model.entity.CopiesEntity;
import dev.ragnarok.fenrir.db.model.entity.Entity;
import dev.ragnarok.fenrir.db.model.entity.EntityWrapper;
import dev.ragnarok.fenrir.model.feedback.FeedbackType;

public class CopyEntity extends FeedbackEntity {

    @SerializedName("copies")
    private CopiesEntity copies;
    @SerializedName("copied")
    private EntityWrapper copied = new EntityWrapper();

    @SuppressWarnings("unused")
    public CopyEntity() {
    }

    public CopyEntity(@FeedbackType int type) {
        setType(type);
    }

    public CopiesEntity getCopies() {
        return copies;
    }

    public CopyEntity setCopies(CopiesEntity copies) {
        this.copies = copies;
        return this;
    }

    public Entity getCopied() {
        return copied.get();
    }

    public CopyEntity setCopied(Entity copied) {
        this.copied = new EntityWrapper().wrap(copied);
        return this;
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/db/model/entity/feedback/CopyEntity.kt
package dev.ragnarok.fenrir.db.model.entity.feedback

import androidx.annotation.Keep
import dev.ragnarok.fenrir.db.model.entity.CopiesEntity
import dev.ragnarok.fenrir.db.model.entity.DboEntity
import dev.ragnarok.fenrir.model.feedback.FeedbackType
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Keep
@Serializable
@SerialName("copy")
class CopyEntity : FeedbackEntity {
    var copies: CopiesEntity? = null
        private set

    private var copied: DboEntity? = null

    @Suppress("UNUSED")
    constructor()
    constructor(@FeedbackType type: Int) {
        this.type = type
    }

    fun setCopies(copies: CopiesEntity?): CopyEntity {
        this.copies = copies
        return this
    }

    fun getCopied(): DboEntity? {
        return copied
    }

    fun setCopied(copied: DboEntity?): CopyEntity {
        this.copied = copied
        return this
    }
}